### PR TITLE
Heatmaps

### PR DESCRIPTION
**Pull request by ryanbaumann**
* Adds support for a heatmap visualization type.  
* Upgrades GL JS to v0.41.0
* Updates Readme
* Adds a new helper util function to create heatmap-weight stops
* Adds heatmap to example notebook

### TO DO

- [x] Add tests for heatmap class
- [x] Add tests for `create_weight_stops` util function
- [x] PR review @perrygeo

### CODE REVIEW

**Comment by ryanbaumann** [comment]
https://github.com/mapbox/mapboxgl-jupyter/pull/16/commits/dde580c6183419cd1c899a54f6b1a7ffc225c8b3#diff-2df5c621fb5e45191036ab5f82a59f9bR60

I could probably abstract this function to apply to both radius stops and weight stops without creating a new utility function.

File: mapboxgl/viz.py [hunk 1]
```diff
@@ -187,3 +187,90 @@ def create_html(self):
...
+                 data,
+                 access_token=None,
+                 center=(0, 0),
+                 weight_property=None,
+                 weight_stops=None,
+                 color_stops=None,
+                 radius_stops=None,
+                 div_id='map',
+                 height='500px',
+                 style_url="mapbox://styles/mapbox/light-v9?optimize=true",
```
Comment (perrygeo) [inline mapboxgl/viz.py]: neat trick, style optimized vector tiles! https://blog.mapbox.com/style-optimized-vector-tiles-39868da81275

Comment (ryanbaumann) [reply mapboxgl/viz.py]: On by default to speed up base map rendering and network tile requests.

**Comment by ryanbaumann** [comment]
Thanks for the review @perrygeo!

**Pull request merged**